pkg/container-utils/docker/docker.go

```diff
@@ -278,3 +266,40 @@ func getContainerImageNamefromImage(image string, containerID string) string {
 	// Case 3 or 4
 	return image
 }
+
+// `getContainerData` takes in two pointers to `dockertypes.ContainerJSON` and `dockertypes.Container`
+// and returns a pointer to a `runtimeclient.ContainerData` and an error. It extracts basic metadata about
+// a Docker container and constructs a `runtimeclient.ContainerData` struct with this information.
+// The function also maps the container state to a runtime client state.
+// If either of the input parameters is `nil`, the function returns an error.
+func getContainerData(containerJSON *dockertypes.ContainerJSON, container *dockertypes.Container) (*runtimeclient.ContainerData, error) {
```

Comment:
IMO, I would prefer this function to receive the ID, Name, Image and State.

Reply:
What about moving the call of `strings.TrimPrefix`, `getContainerImageNamefromImage` and `containerStatusStateToRuntimeClientState` inside `buildContainerData`?

Reply:
That's indeed a very good idea! Done. Thanks!